rest-client/src/main/java/org/sagebionetworks/bridge/rest/UserSessionInfoProvider.java

```diff
@@ -68,14 +76,17 @@ public synchronized UserSessionInfo retrieveSession() throws IOException {
         return session;
     }
     
-    synchronized void setSession(UserSessionInfo newSession) {
+    synchronized void setSession(final UserSessionInfo newSession) {
         if (LOG.isDebugEnabled()) {
             String token = (newSession==null) ? "null" : newSession.getSessionToken();
             LOG.debug(debugString(token));
         }
 
         mergeReauthToken(this.session, newSession);
         this.session = newSession;
+        for(UserSessionInfoChangeListener listener : changeListeners) {
+            listener.onChange(this.session);
```

Comment:
If a listener throws an exception, what should happen? Should you call the rest of the listeners?

Reply:
I'll log the exception and continue. They shouldn't affect each other. Not sure what other action to take